spec/controllers/users_controller_spec.rb

```diff
@@ -1,5 +1,123 @@
 require 'rails_helper'
 
 RSpec.describe UsersController, type: :controller do
-
+  before do
+    stub_request(:any, /localhost:9250/)
+  end
+  
+  describe 'GET index' do
+    context 'without signing in' do
+      it 'redirects to sign in' do
+        get :index
+        expect(response).to redirect_to('/users/sign_in')
+      end
+    end
+    
+    context 'signed in user' do
+      let(:author) { FactoryGirl.create(:user) }
+      let(:maintainer) { FactoryGirl.create(:user) }
+      
+      it 'returns users' do
+        sign_in author
+        
+        get :index
+        expect(assigns(:users)).to eq([author, maintainer])
+      end
+    end
+  end
+  
+  describe 'GET show' do
+    let(:author) { FactoryGirl.create(:user) }
+    let(:maintainer) { FactoryGirl.create(:user) }
+    let(:lone_wolf) { FactoryGirl.create(:user) }
+    
+    let(:dataset_a) { FactoryGirl.create(:dataset, author: author) }
+    let(:dataset_b) { FactoryGirl.create(:dataset, maintainer: author) }
+    let(:dataset_c) { FactoryGirl.create(:dataset, author: author, maintainer: maintainer) }
+    let(:dataset_d) { FactoryGirl.create(:dataset, maintainer: author, author: maintainer) }
+    
+    let(:dataset_w) { FactoryGirl.create(:dataset, author: maintainer, visibility: Dataset::PRIVATE) }
+    let(:dataset_x) { FactoryGirl.create(:dataset, maintainer: maintainer, visibility: Dataset::PRIVATE) }
+    let(:dataset_y) { FactoryGirl.create(:dataset, author: maintainer, maintainer: author, visibility: Dataset::PRIVATE) }
+    let(:dataset_z) { FactoryGirl.create(:dataset, maintainer: maintainer, author: author, visibility: Dataset::PRIVATE) }
+    
+    context 'without signing in' do
+      it 'redirects to sign in' do
+        get :show, params: {id: author.id}
+        expect(response).to redirect_to('/users/sign_in')
+      end
+    end
+    
+    context 'signed in user' do
+      describe 'viewing self as author' do
+        it 'assigns correct profile and correct datasets' do
+          sign_in author
+          get :show, params: {id: author.id}
+          expect(assigns(:user)).to eq(author)
+          expect(assigns(:user)).to_not eq(maintainer)
+          expect(assigns(:user_datasets)).to eq([dataset_a, dataset_b, dataset_c, dataset_d, dataset_y, dataset_z])
```

Comment:
You can use `match_array` (I think) to compare arrays. This would save you the trouble of checking for length and what's not in the array.

Reply:
ahh okay, yea that's exactly what i was trying to test for. using `eq` would match parts of of the array.

Reply:
`match_array` was similar to eq, only checking if array included some items. updated some tests using `contain_exactly` to check exactly whats in the array regardless of order.

Reply:
Docs say they're equivalent with different syntax:
https://relishapp.com/rspec/rspec-expectations/v/3-7/docs/built-in-matchers/contain-exactly-matcher

Reply:
you're right, they're same. i discovered a different issue instead.